truss-train/truss_train/definitions.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict, List, Optional, Union
+
+from pydantic import Field
+from truss.base import truss_config
+from truss.base.custom_types import SafeModel
+
+
+class SecretReference(SafeModel):
+    name: str
+
+
+class Compute(SafeModel):
+    node_count: int = 1
+    cpu_count: int = 1
+    memory: str = "2Gi"
+    accelerator: Optional[truss_config.AcceleratorSpec] = None
+
+
+class Runtime(SafeModel):
+    start_commands: List[str] = []
+    environment_variables: Dict[str, Union[str, SecretReference]] = {}
+
+
+class Image(SafeModel):
+    base_image: str
+
+
+class TrainingJob(SafeModel):
+    image: Image
+    compute: Compute = Compute()
+    runtime: Runtime = Runtime()
+
+
+class TrainingProject(SafeModel):
```

Comment:
I intentionally left off the `blob` stuff for now, I think it'll be easier to add that in targeted followups. I imagine we'll eventually zip up the directory and pass it through to server to upload to an S3 bucket of our choosing